### ISSUE TITLE
Exception thrown in Constructor of CookieContainer

### ISSUE THREAD

**Issue by nikeee** [dotnet/corefx#7548]
I've got an issue with the CookieContainer in my UWP application.

### Expected behaviour
The constructors of `CookieContainer`, `HttpClient` and `HttpClientHandler` should not thow an exception when invoked without a parameter.

### Actual behavior
- Creating a new instance of `CookieContainer` throws an exception.
- Creating a new `HttpClient` or `HttpClientHandler` throws an exception because they create a new `CookieContainer` internally.
- It is not possible to use any of these classes.

Stack trace/message of the exception:
```
System.ArgumentException: The parameter is incorrect.

hostName
   at Windows.Networking.Connectivity.NetworkInformation.GetHostNames()
   at System.Net.NetworkInformation.HostInformation.GetFixedInfo()
   at System.Net.NetworkInformation.HostInformation.get_FixedInfo()
   at System.Net.NetworkInformation.HostInformation.get_DomainName()
   at System.Net.CookieContainer..ctor()
   at System.Net.Http.HttpClientHandler..ctor()
```
The last function in the call stack:
[`NetworkInformation.GetHostNames`](https://msdn.microsoft.com/library/windows/apps/windows.networking.connectivity.networkinformation.gethostnames), which doesn't take any parameters. Therefore, I don't which parameter `hostName` could be wrong that I am passing causing the exception.

### Steps to fix/reproduce
I figured out how to work around the problem:
- Goto `Control Panel\All Control Panel Items\System` -> `Advanced System Settings` -> `Computer Name` -> `Change` -> `More`
- Clear what's in the text box of "Primary DNS suffix of this computer" (it was `42` before, since this is my local dns suffix, so the FQDN of my host is `hostname.42`).
- This is reproducible on my machine. Didn't test this on other machines.
- It works using a DNS suffix which is not a number (tested with `foo`).

If this behaviour is intended, the possibility of an exception when calling `NetworkInformation.GetHostNames` should be documented.

Windows Build: Version 1511 (10586.164)

Edit:
What I think is that NetworkInformation.GetHostNames() calls `GetAllHostNames()` from `Windows.Networking.HostName.dll` (listed as import of the dll of NetworkInformation). The `HostName` constructor is then used to create the instances of `HostName`. I think the call to the function `HostNameImpl::CanonicalizeHostName` fails, since it raises an exception when it fails. Maybe it's [`DnsValidateName`](https://msdn.microsoft.com/en-us/library/windows/desktop/ms682032%28v=vs.85%29.aspx) that classifies the host name `host.42` as invalid (which seems correct to me).

**Comment by davidsh** [comment]
What is the hostname of your client machine?  Does it use any non-ASCII characters?

We are aware of a bug in the Windows 10 OS where `Windows.Networking.Connectivity.GetHostNames` will return `ArgumentException` in some cases when the hostname contains non-ASCII characters. This is probably the root cause of your issue.

This bug has been fixed in the Windows 10 OS and the fix will appear in the WIndows 10 Anniversary Update later this year.  Your currently running build 10586 which still has this bug.

**Comment by nikeee** [comment]
> Does it use any non-ASCII characters?

The hostname was `cornflake.42` (including DNS suffix), so no non-ASCII chars involved.
I am assuming that Windows didn't like the `42` as a domain. Also, the `ping` command was not able to resolve other hosts in the network, only if ping was used on things like `otherhost.` (including the `.`, but without the `42`). Since `nslookup otherhost.42` worked flawlessly, this seemed pretty strange to me.

No host name in the network contains non-ASCII chars.
I changed the local DNS suffix to a 3-letter-lower-case one and it woks - using a number was a silly idea anyway.

**Comment by davidsh** [comment]
In your specific case, using a DNS suffix label starting with a numeral is not allowed. DNS suffix labels need to start with a letter. So, the error returned is by-design.

**Comment by davidsh** [comment]
cc: @stephentoub

**Comment by nikeee** [comment]
> So, the error returned is by-design.

I do understand that this is intended behaviour. At least it should be documented that an exception is thrown when using a `CookieContainer`. I had no idea that a CookieContainer internally calles `GetAllHostNames` and then throws an exception because of an invalid client host name. When I looked up the code of the CookieContainer, I ended up in [HostInformationPal.Windows.cs](https://github.com/dotnet/corefx/blob/master/src/Common/src/System/Net/NetworkInformation/HostInformationPal.Windows.cs), trying to trace the call stack of the exception. But there wasn't even a call to `NetworkInformation.GetHostNames`.

Also, neither the [Constructor of `HostName`](https://msdn.microsoft.com/en-us/library/windows/apps/br207118) nor [`NetworkInformation.GetHostNames`](https://msdn.microsoft.com/en-us/library/windows/apps/windows.networking.connectivity.networkinformation.gethostnames) state that there is an exception in my special case. Only "The constructor will fail if the hostName parameter is null or contains an empty string.".
